code/game/objects/structures/beds_chairs/bed.dm

```diff
@@ -52,29 +52,41 @@
 	icon_state = "down"
 	anchored = FALSE
 	resistance_flags = NONE
-	var/foldabletype = /obj/item/roller
+	/// Internal storage for the roller item that created us
+	var/obj/item/roller
```

Comment:
you dont register for this being qdeleted

Reply:
Please suggest so I can web commit, at work tyly